scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -843,13 +715,20 @@ class ScioContext private[scio] (val options: PipelineOptions,
    * @param io     an implementation of `ScioIO[T]` trait
    * @param params configurations need to pass to perform underline read implementation
    */
-  def read[T: ClassTag](io: ScioIO[T])(params: io.ReadP): SCollection[T] = requireNotClosed {
-    if (this.isTest) {
-      this.getTestInputNio(io.id)
-    } else {
-      io.read(this, params)
+  def read[T: ClassTag](io: ScioIO[T])(params: io.ReadP): SCollection[T] =
+    readImpl[T](io)(params)
+
+  private def readImpl[T: ClassTag](io: ScioIO[T])(params: io.ReadP): SCollection[T] =
+    requireNotClosed {
+      if (this.isTest) {
+        this.getTestInputNio(io.id)
+      } else {
+        io.read(this, params)
+      }
     }
-  }
+
+  def read[T: ClassTag](io: ScioIO[T]{ type ReadP = Unit }): SCollection[T] =
```

Comment:
Pure syntactic sugar. If the IO does not require parameters, you can ignore the second parameters block.